tests/test_install_in_fresh_environment.py

```
@@ -0,0 +1,625 @@
+        return sorted(packages)
+
+    core_packages = process_imports(core_imports)
+    dev_packages_all = process_imports(dev_imports)
+
+    # Remove core packages from dev (dev should only have additional ones)
+    dev_only_packages = [pkg for pkg in dev_packages_all if pkg not in core_packages]
+
+    return core_packages, dev_only_packages
+
+
+def run_command(cmd: List[str], cwd: str = None, timeout: int = 300) -> Tuple[int, str, str]:
```

Ruff (0.14.10)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)